src/components/pageNav/index.jsx

```diff
@@ -33,13 +55,35 @@ class PageNav extends React.PureComponent {
     }));
   };
 
+  handleShowCompletedAlert = () => {
+    const { allAnswered } = this.state;
+
+    if (!allAnswered) {
+      this.setState(prevState => ({
+        ...prevState,
+        showCompletedAlert: true,
+      }));
+    }
+  };
+
   handleHideAlert = () => {
     this.setState(prevState => ({
       ...prevState,
       showAlert: false,
     }));
   };
 
+  handleHideCompletedAlert = () => {
+    const { allAnswered } = this.state;
+
+    if (allAnswered) {
+      this.setState(prevState => ({
+        ...prevState,
+        showCompletedAlert: false,
+      }));
+    }
+  };
+
```

Comment:
Should closing the notification depend on the `allAnswered` state?  Is this method ever called when `allAnswered` could be `false` (i.e. wouldn't the notification only be visible if `allAnswered` is `true`)?

Reply:
This was a strange bug that I came across. I had to implement this in order to ensure that the render method didn't toggle to `false` immediately after toggling to `true`. I was console logging the value of `showCompletedAlert` and it showed to logs (from within the `render()` method) when trying to trigger the alert. Both fired one after the other. The first updated `showCompletedAlert` to `true` and the second reverted back to `false`. I couldn't think of any other way to handle this.